src/services/codecovEventMetrics/useStoreCodecovEventMetric.tsx

```diff
@@ -0,0 +1,103 @@
+import { useMutation } from '@tanstack/react-query'
+import { useParams } from 'react-router-dom'
+
+import Api from 'shared/api'
+
+const mutationQuery = `
+  mutation storeEventMetric(
+    $input: StoreEventMetricsInput!
+  ) {
+    storeEventMetric(
+      input: $input
+    ) {
+      error {
+        ... on UnauthenticatedError {
+          message
+        }
+        ... on ValidationError {
+          message
+        }
+      }
+    }
+  }
+`
+
+interface Params {
+  provider: string
+}
+
+interface StoreEventMetricMutationArgs {
+  owner: string
+  event: string
```

Comment:
making event a union of string literals would be helpful IMO. It was nice having the type-safety with @nicholas-codecov's Sentry metrics wrapper. Not 0 work though, so could be convinced we don't need this at this point.

Reply:
There is some validation at the [service level](https://github.com/codecov/shared/blob/f9c0acde3cfc54c72f42f812b0212a5066429e89/shared/django_apps/codecov_metrics/service/codecov_metrics.py). 

The reason I did not add types here is because we will probably add more metrics in the future, and my thinking is it more upfront maintenance to have parity between the service and here. 

Happy to revisit at a later date when we have a comprehensive list that's finalized.

Reply:
Fair enough! Could we return a 4xx on bad event type instead of just logging a warning? This way the constraints are strictly enforced and we can't end up with random stuff in the DB. Also makes it clear that you've made a typo in the FE when setting up metrics 😅

Reply:
Oh I totally missed the early `return` in the bad event type path. That works.

Yea I think it would be great to have the type safety here eventually. I'm happy with you making an issue and returning to this later! 👍

Reply:
https://github.com/codecov/engineering-team/issues/1998